### PR TITLE
Implemented Password Pawned Checker

### PR DESCRIPTION
**Pull request by naveennivas**
**Classes Created** 
1. PasswordPawnedOperationsImpl  --> _under actions_ (with the interface PasswordPawnedOperations)
          -> Implemented the method _pawned checker_ from the interface.
          -> The method makes use of the pawned checker api from github to check whether the password has been pawned or not.
          -> Returns true(leaked) or false (not leaked) to the object passwordpawnedoutput in the class _PasswordPawnedOutput_
          -> The status variable in the output method is also set here.

2. PasswordPawnedChecker -->  _under api_
           -> Input parameter password is received to string _pass_ then this is passed to the method on to _PasswordPawnedOperationsImpl_ to check whether its leaked or not.

3. PasswordPawnedOutput --> _under model_
            -> has a method called _PasswordPawnedOutput_ which contains a boolean _status_ variable to say true(leaked) or false(not leaked/safe).
 **Edits made in _POM.xml_**
1. added dependencies  from _https://www.twilio.com/blog/2018/06/round-up-libraries-for-checking-pwned-passwords-in-your-7-favorite-languages.html_  , api for pawned checker
2. added dependencies  from _https://mvnrepository.com/artifact/com.squareup.okhttp3/okhttp_ to use the api pawned checker

### CODE REVIEW

File: src/main/java/com/psipher/application/api/PasswordPawnedChecker.java [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+@RestController
+public class PasswordPawnedChecker {
+    private boolean status;
+    PasswordPawnedOperations passwordpawnedoperations;
+    @Autowired
+    public PasswordPawnedChecker(PasswordPawnedOperations passwordpawnedoperations) {
+        this.passwordpawnedoperations = passwordpawnedoperations;
+    }
+    @RequestMapping(path = "/passwordpawnedchecker", method = RequestMethod.GET)
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/api/PasswordPawnedChecker.java:20]: It should not be a get request

File: src/main/java/com/psipher/application/actions/PasswordPawnedOperationsImpl.java [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+import org.springframework.stereotype.Component;
+
+/**
+ * Implements PasswordPawnedCheckerOperations and provide all functionality mentioned in the interface.
+ */
+@Component
+public class PasswordPawnedOperationsImpl implements PasswordPawnedOperations {
+
+    @Override
+    public PasswordPawnedOutput passwordpawnedchecker(String pass) {
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/PasswordPawnedOperationsImpl.java:17]: We should not reuse the model which was used in rest output request. For internal model create a new model or just pass a boolean. We dont need extra model here

File: src/main/java/com/psipher/application/actions/PasswordPawnedOperationsImpl.java [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        PasswordPawnedOutput passwordpawnedoutput = new PasswordPawnedOutput();
+        PwnedPasswordClient client = new PwnedPasswordClient(new OkHttpClient(), "https://api.pwnedpasswords.com/range", "");
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/PasswordPawnedOperationsImpl.java:19]: Move this url https://api.pwnedpasswords.com/range to resource or config and url and extract from that place

File: src/main/java/com/psipher/application/actions/PasswordPawnedOperationsImpl.java (hunk 2 above)
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/PasswordPawnedOperationsImpl.java:17]: NIT: change from passwordpawnedchecker --> passwordPawnedChecker

File: src/main/java/com/psipher/application/actions/PasswordPawnedOperations.java [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+
+import com.psipher.application.model.PasswordPawnedOutput;
+
+public interface PasswordPawnedOperations {
+    /**
+     * It checks whether the password has been pawned for the user.
+     * @param pass the input password of the user
+     * @return pawned or not
+     */
+    PasswordPawnedOutput passwordpawnedchecker(String pass);
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/PasswordPawnedOperations.java:11]: Change output to boolean why do we need to depend on rest output object

File: src/main/java/com/psipher/application/actions/PasswordPawnedOperations.java [hunk 5]
```diff
@@ -0,0 +1,12 @@
+package com.psipher.application.actions;
+
+import com.psipher.application.model.PasswordPawnedOutput;
+
+public interface PasswordPawnedOperations {
+    /**
+     * It checks whether the password has been pawned for the user.
+     * @param pass the input password of the user
+     * @return pawned or not
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/actions/PasswordPawnedOperations.java:9]: NIT: @return true/false

File: src/main/java/com/psipher/application/api/PasswordPawnedChecker.java [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestMethod;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+
+@RestController
+public class PasswordPawnedChecker {
+    private boolean status;
+    PasswordPawnedOperations passwordpawnedoperations;
```
Comment (rahulkrishnan221) [inline src/main/java/com/psipher/application/api/PasswordPawnedChecker.java:15]: make this private

**Review by rahulkrishnan221 — CHANGES_REQUESTED** [review]

**Comment by naveennivas** [comment]
Have made the changes requested.

**Review by rahulkrishnan221 — APPROVED** [review]

**Pull request merged**